### PR TITLE
fix: resolve TODOs about not throwing errors - WPB-5140

### PR DESCRIPTION
**Pull request by caldrian**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-5140" title="WPB-5140" target="_blank">TaskWPB-5140</a>  [iOS] Refactor user session creation to handle async core crypto setup
  </td></table>
  <br />
 

### Issue

We were some cases swallowing errors, which can hide errors and makes it impossible to do anything error recovery at the top level.

I've also added tickets reference to TODO which were missing it.

---

### Checklist

- [ ] Title contains a reference JIRA issue number like `[WPB-XXX]`.
- [ ] Description is filled and free of optional paragraphs.
- [ ] Adds/updates automated tests.

### CODE REVIEW

File: wire-ios-sync-engine/Source/UserSession/ZMUserSession/ZMUserSession.swift [hunk 1]
```diff
@@ -771,7 +771,7 @@ extension ZMUserSession: ZMNetworkStateDelegate {
 // MARK: - UpdateEventProcessor
 
 // swiftlint:disable todo_requires_jira_link
-// TODO: [jacob] find another way of providing the event processor to ZMissingEventTranscoder
+// TODO: [WPB-9089] find another way of providing the event processor to ZMissingEventTranscoder
```
Comment (markusfassbender) [inline wire-ios-sync-engine/Source/UserSession/ZMUserSession/ZMUserSession.swift:761]: @typfel there are quite some of these comments in code, would be nice to convert all of these in future into tickets 🙂

**Review by markusfassbender — APPROVED** [review]

File: wire-ios-sync-engine/Source/UserSession/ZMUserSession/ZMUserSession.swift (hunk 1 above)
Comment (caldrian) [reply wire-ios-sync-engine/Source/UserSession/ZMUserSession/ZMUserSession.swift:761]: I agree, eventually the SwiftLint exceptions should be removed.

**Review by samwyndham — APPROVED** [review]
Nice work. I think anything we can do to surface errors or bad state is super worthwhile!

**Review by netbe — APPROVED** [review]
resolved conflict

**Pull request merged**